streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -432,6 +437,10 @@ protected void maybeCommit() {
     private void commitAll() {
         for (StreamTask task : activeTasks.values()) {
             commitOne(task, time.milliseconds());
+            final ProcessorNode node = task.node();
```

Comment:
Is it correct to do the punctuation sensor recording for task.node() or even here? Why not do it in `StreamTask's` `punctuate`?